modules/tinymce/src/core/test/ts/browser/content/EditorContentTest.ts

```diff
@@ -291,4 +309,112 @@ describe('browser.tinymce.core.content.EditorContentTest', () => {
+        const editor = hook.editor();
+        editor.setContent(unsanitizedHtml);
+        TinyAssertions.assertContent(editor, sanitizedHtml);
+      });
+
+      it('getContent html with unsanitized content should get sanitized html', () => {
+        const editor = hook.editor();
+        editor.setContent(unsanitizedHtml);
+        const content = editor.getContent();
+        assert.equal(content, sanitizedHtml, 'Unsanitized html should be sanitized');
+      });
+
+      it('getContent text with unsanitized content should get text from sanitized content', () => {
+        const editor = hook.editor();
+        editor.setContent(unsanitizedHtml);
+        const text = editor.getContent({ format: 'text' });
+        assert.equal(text, 'XSS', 'Text should be retrieved from sanitized html');
+      });
+
+      it('getContent tree with unsanitized content should get sanitized tree', () => {
+        const editor = hook.editor();
+        editor.setContent(unsanitizedHtml);
+        assertContentTreeEqualToHtml(editor, sanitizedHtml, 'Unsanitized html should be sanitized');
+      });
+
+      it('setContent tree with content altered in BeforeSetContent should set sanitized html', () => {
+        const editor = hook.editor();
+        testSetContentTreeWithContentAlteredInBeforeSetContent(editor, unsanitizedHtml, 'Replaced content should be sanitized', sanitizedHtml);
+      });
+
+      it('getContent tree with content altered in GetContent should get sanitized html', () => {
+        const editor = hook.editor();
+        testGetContentTreeWithContentAlteredInGetContent(editor, unsanitizedHtml, 'Replaced content should be sanitized', sanitizedHtml);
+      });
+    });
+
+    context('TINY-9600: Test unsanitized content with inline: ' + options.inline + ' and xss_sanitization: false', () => {
+      const hook = TinyHooks.bddSetupLight<Editor>({
+        base_url: '/project/tinymce/js/tinymce',
+        xss_sanitization: false,
+        ...options
+      }, []);
+
+      it('setContent with unsanitized content should set content unaltered', () => {
+        const editor = hook.editor();
+        editor.setContent(unsanitizedHtml);
+        TinyAssertions.assertContent(editor, unsanitizedHtml);
+      });
+
+      it('getContent html with unsanitized content should get content unaltered', () => {
+        const editor = hook.editor();
+        editor.setContent(unsanitizedHtml);
+        const content = editor.getContent();
+        assert.equal(content, unsanitizedHtml, 'Unsanitized html should not be altered');
+      });
+
+      it('getContent html with iframe with child node should get the content as expected and not error', () => {
+        const editor = hook.editor();
+        editor.setContent('<p><iframe><p>test</p></iframe></p>');
+        const content = editor.getContent();
+        assert.equal(content,
+          PlatformDetection.detect().browser.isSafari()
```

Comment:
This is pretty strange this should be investigated in a follow up. If the editors don't produce the same HTML on save it could produce very strange errors if say you diff the initial value with the current value and you made no changes depending on platform it would be dirty vs not dirty. This should be normalized in the parser/serializer code. Maybe with a custom filter. Log a follow up ticket.

- [ ] Log ticket to investigate safari specific html output

Reply:
Created ticket TINY-9624 to investigate Safari-specific HTML output.